presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java

```diff
@@ -0,0 +1,284 @@
+        KeyStore keystore;
+
+        try {
+            // Attempt to read the key store as a PEM file
+            // For PEM encoded keys, the password is used to decrypt the specific key
+            log.debug("Attempting to load keystore as PEM format");
+            keystore = PemReader.loadKeyStore(keystoreFile, keystoreFile, keystorePassword);
+            log.debug("Successfully loaded keystore as PEM format");
+        }
+        catch (IOException | GeneralSecurityException e) {
+            log.debug("Failed to load keystore as PEM, attempting JKS format: {}", e.getMessage());
+            keystore = getInstance(getDefaultType());
+            try (InputStream in = new FileInputStream(keystoreFile)) {
+                keystore.load(in, keystorePassword.map(String::toCharArray).orElse(null));
+            }
+            log.debug("Successfully loaded keystore as JKS format");
+        }
+
+        validateCertificates(keystore);
+        return keystore;
+    }
+
+    private KeyManager[] createKeyManagers(KeyStore keystore) throws GeneralSecurityException
+    {
+        char[] keyManagerPassword = keystorePassword.map(String::toCharArray).orElse(new char[0]);
+        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(getDefaultAlgorithm());
+        keyManagerFactory.init(keystore, keyManagerPassword);
+        return keyManagerFactory.getKeyManagers();
+    }
+
+    private X509TrustManager createTrustManager(KeyStore truststore) throws GeneralSecurityException
+    {
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(getDefaultAlgorithm());
+
+        // If no custom truststore is provided, use system default (pass null)
+        // If a custom truststore is provided but empty, that's an error
+        if (truststore != null) {
+            try {
+                // Check if truststore has any certificates
+                List<String> aliases = Collections.list(truststore.aliases());
+                if (aliases.isEmpty()) {
+                    throw new GeneralSecurityException("Truststore is empty - no trusted certificates found");
+                }
+                log.debug("Truststore contains {} certificate(s): {}", aliases.size(), aliases);
+            }
+            catch (KeyStoreException e) {
+                throw new GeneralSecurityException("Failed to read truststore", e);
+            }
+        }
+
+        trustManagerFactory.init(truststore);
+
+        TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+        if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
+            throw new RuntimeException("Unexpected default trust managers: " + Arrays.toString(trustManagers));
+        }
+
+        return (X509TrustManager) trustManagers[0];
+    }
+
+    private static KeyStore loadTrustStore(File trustStorePath, Optional<String> trustStorePassword)
+            throws IOException, GeneralSecurityException
```

Comment:
IOException is not getting thrown and is no more a checked exception for this method. Lets remove it from method signature.